app/presenters/profile.rb

```diff
@@ -15,6 +16,21 @@ def exercises
         @exercises ||= user.exercises.where(archived: false).where('iteration_count > 0')
       end
 
+      def progress
+        bag = Hash.new 0
+        user.exercises.each{|exe| bag[exe.problem.language] += 1}
+        bag
+        tracks = track_count
+        bag.each{|k, v| bag[k] = [v, tracks[k]]}
```

Comment:
So this is re-using the same hash and putting different data in it? Maybe this whole progress thing should be its own little object so we can spread things out a little bit.